contracts/bithive/src/view.rs

```
@@ -248,8 +264,12 @@ impl Contract {
             input_to_sign.previous_output.vout.into(),
         );
 
-        if account.pending_sign_psbt.is_some() {
-            verify_sign_withdrawal_psbt(account.pending_sign_psbt.as_ref().unwrap(), &psbt);
+        if let Some(pending_sign_psbt_idx) = pending_sign_psbt_idx {
+            let pending_sign_psbt = account
+                .pending_sign_psbts
+                .get(pending_sign_psbt_idx)
```

Should we use `expect` here to panic with more meaningful error if the index is invalid?

---

updated